notebooks/IO_adapters.ipynb

```diff
@@ -0,0 +1,389 @@
+{
+  "nbformat": 4,
+  "nbformat_minor": 0,
+  "metadata": {
+    "colab": {
+      "name": "IO_adapters.ipynb",
+      "provenance": [],
+      "collapsed_sections": []
+    },
+    "kernelspec": {
+      "name": "python3",
+      "display_name": "Python 3"
+    }
+  },
+  "cells": [
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "LVLE6fzffmJk",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "import tensorflow as tf\n",
+        "from tensorflow import keras\n",
+        "from tensorflow.keras import layers\n",
+        "\n",
+        "import numpy as np\n",
+        "import unittest"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "EYCSozXYhIvR",
+        "colab_type": "text"
+      },
+      "source": [
+        "# Input adapter\n",
+        "This model transforms the input image into tenfor of desired shape (in our setup 16x16x40)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "nJlxo9f7f5BP",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "def create_input_adapter(input_shape, size=16, depth=40, activation=None):\n",
+        "  \"\"\"Creates an input adapter module for the input image.\n",
+        "  The input adapter transforms input image of given shape \n",
+        "  into a tensor of target shape.\n",
+        "\n",
+        "  Arguments: \n",
+        "    input_shape: shape of input image (HxWxC). Image width and height \n",
+        "      must be devidible by size. H*W*C must be less than or equal\n",
+        "      to size*size*depth.\n",
+        "\n",
+        "    size: height and width of the output tensor after space2depth operation. \n",
+        "\n",
+        "    depth: number of channels in the output tensor.\n",
+        "\n",
+        "    activation: conv layer activation function\"\"\"\n",
+        "  def _get_parameters():\n",
+        "    # TODO(Dina): Google Python guidelines request that you use space everywhere\n",
```

Comment:
This comment can probably be removed (also check if there are other ones like this).